### PR TITLE
🌱   Add Machinepool conversion

### LINKED ISSUE #5265: Add conversion for MachinePool
Let's add the following for MachinePools (after #5248)
* conversion.go in v1alpha4 package
* conversion fuzzy test
* webhook conversion patches in config/

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
Adds MachinePool conversion. 

WIP until #5248 has been merged

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5265

### CODE REVIEW

File: exp/api/v1alpha3/conversion.go [hunk 1]
```diff
@@ -17,26 +17,72 @@ limitations under the License.
...
+
+func (src *MachinePool) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta1.MachinePool)
+
+	if err := Convert_v1alpha3_MachinePool_To_v1beta1_MachinePool(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Preserve Hub data on conversion except for metadata
+	if err := utilconversion.MarshalData(src, dst); err != nil {
```
Comment (sbueringer) [inline exp/api/v1alpha3/conversion.go:54]: Not sure if the annotation is meant to be used for conversions in this "direction" (spoke=>hub). 
In all other cases we currently have we implement it for hub=>spoke (i.e. in ConvertFrom)

I think this should work as long as we don't try to do it for the same resource in both directions at the same time.

File: exp/api/v1alpha3/conversion.go [hunk 2]
```diff
@@ -17,26 +17,72 @@ limitations under the License.
...
 
-func Convert_v1alpha3_MachinePool_To_v1beta1_MachinePool(in *MachinePool, out *v1beta1.MachinePool, s conversion.Scope) error {
+func Convert_v1alpha3_MachinePool_To_v1beta1_MachinePool(in *MachinePool, out *v1beta1.MachinePool, s apimachineryconversion.Scope) error {
 	if err := autoConvert_v1alpha3_MachinePool_To_v1beta1_MachinePool(in, out, s); err != nil {
 		return err
 	}
 
 	// Replace v1alpha3 finalizer to allow old MachinePools to get deleted.
-	if !controllerutil.ContainsFinalizer(out, MachinePoolFinalizer) {
+	if controllerutil.ContainsFinalizer(out, MachinePoolFinalizer) {
```
Comment (fabriziopandini) [inline exp/api/v1alpha3/conversion.go:38]: Do we need to do the reverse operation when converting from v1beta1 to v1alpha3
@CecileRobertMichon to confirm about this and the changes in the expression

File: exp/api/v1alpha3/conversion_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (killianmuldoon) [inline exp/api/v1alpha3/conversion_test.go:2]: nit update year

File: exp/api/v1alpha3/conversion.go (hunk 2 above)
Comment (CecileRobertMichon) [reply exp/api/v1alpha3/conversion.go:38]: yes, I think so.. In practice this won't be used much (I don't foresee many users doing beta to alpha3 conversion for MachinePool and manually changing the group) but we should cover it.

Comment (CecileRobertMichon) [reply exp/api/v1alpha3/conversion.go:38]: thanks for fixing the logic, that looks correct to me

File: exp/api/v1alpha3/conversion.go (hunk 1 above)
Comment (sbueringer) [reply exp/api/v1alpha3/conversion.go:54]: @vincepri WDYT?

File: exp/api/v1alpha3/conversion.go (hunk 2 above)
Comment (sbueringer) [reply exp/api/v1alpha3/conversion.go:38]: Should be fixed now, PTAL

File: exp/api/v1alpha3/conversion.go (hunk 1 above)
Comment (sbueringer) [reply exp/api/v1alpha3/conversion.go:68]: Talked to vince. We should never add the conversion data / annotation in this direction because then it's stored in etcd, which is something we don't want.

File: bootstrap/kubeadm/types/upstreamv1beta1/conversion_test.go [hunk 4]
```diff
@@ -32,27 +32,31 @@ func TestFuzzyConversion(t *testing.T) {
 		Spoke: &ClusterConfiguration{},
 		// NOTE: Kubeadm types does not have ObjectMeta, so we are required to skip data annotation cleanup in the spoke-hub-spoke round trip test.
 		SkipSpokeAnnotationCleanup: true,
+		SkipHubAnnotationCleanup:   true,
```
Comment (vincepri) [inline bootstrap/kubeadm/types/upstreamv1beta1/conversion_test.go:35]: Why is this needed?

Comment (sbueringer) [reply bootstrap/kubeadm/types/upstreamv1beta1/conversion_test.go:35]: Forgot to drop it :/

**Comment by sbueringer** [comment]
@vincepri should be fixed, ptal

**Review by vincepri — APPROVED** [review]
/approve
/lgtm
